src/main/java/com/github/lzyzsd/circleprogress/DonutProgress.java

```diff
@@ -84,6 +99,34 @@ public void setProgress(float progress) {
 		}
 	}
 
+	public void setProgress(long bytesRead, long totalBytes) {
+
+		if(this.internalState != Type.SIZE_PROGRESSBAR) {
+
+			mbStatus = new Paint();
+			mbStatus.setColor(finishedStrokeColor);
+			mbStatus.setStyle(Paint.Style.FILL);
+			mbStatus.setStrokeWidth(unfinishedStrokeWidth);
+			mbStatus.setTextSize(40);
+			mbStatus.setTextAlign(Paint.Align.CENTER);
+			mbStatus.setAntiAlias(true);
+
+			this.internalState = Type.SIZE_PROGRESSBAR;
+
+			this.totalBytes = totalBytes;
+		}
+
+
+		this.bytesRead = bytesRead;
+
+		float progress = ((float)((1000 * bytesRead) / totalBytes)) / 1000;
+
+		if(Math.abs(progress - this.progress) > 0.0001) {
```

Comment:
Good thinking, although we probably don't even need updates this frequently (e.g. `0.0001` could be changed to `0.01`).